snap/integrity/dmverity/veritysetup.go

```diff
@@ -0,0 +1,76 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2023 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package dmverity
+
+import (
+	"fmt"
+	"os/exec"
+	"strings"
+
+	"github.com/snapcore/snapd/logger"
+	"github.com/snapcore/snapd/osutil"
+)
+
+// Info represents the dm-verity related data that:
+// 1. are not included in the superblock which is generated by default when running
+//    veritysetup.
+// 2. need their authenticity verified prior to loading the integrity data into the
+//    kernel
+//
+// For now, since we are keeping the superblock as it is, this only includes the root hash.
+type Info struct {
+	RootHash string `json:"root-hash"`
+}
+
+func getRootHashFromOutput(output []byte) (string, error) {
+	rootHash := ""
+
+	for _, line := range strings.Split(string(output), "\n") {
```

Comment:
we tend to use bufio.Scanner for this kind of situtation, you can look around in the code base for that

Reply:
Not sure what to do here. I found 2 different ways to execute commands and get their outputs in the code base:
1) `Cmd.StdoutPipe()` which returns `io.ReadCloser`, that we then use a `bufio.Scanner` to parse (used in `squashfs.go` for example)
2) `Cmd.CombinedOutput()` which returns `[]byte` and felt more appropriate (and simpler) in this case that the output of `veritysetup format` is small in size.

Should I change it to use `StdoutPipe()` and pass a reader to `getRootHashFromOutput()`? Are there any advantages of doing it like that?

Reply:
afaict we tend to use the former  (see ./snap/squashfs/squashfs.go Walk or interfaces/hotplug/udevadm.go), but anyway you can use the Scanner even on []byte by using a bytes.Buffer which is a reader